### DISCUSSION TITLE
Cross platform

### DISCUSSION THREAD

**Discussion by GanAHE** [pelicanmapping/rocky#52] [Q&A / Support] [answered]
Does Rocky plan to support cross-platform development? OsgEarth finds this challenging due to its reliance on OpenGL/OpenGLES, but Vulkan makes it easier to achieve.

**Comment by gwaldron** [comment]
Hello!
Rocky is indeed cross-platform, though we've only run it on Windows and Ubuntu Linux so far.
What platform(s) do you need to target?

(update: I did make some Linux compile fixes today that it needed)

**Reply by GanAHE** [reply]
Thanks for your job!
I think Rocky can work on mainstream Windows\Linux\Android\IOS platforms that will make it widely used by people. 
I'm doing some patch work on the Mobile platform now and successfully compiling Rocky to the Android platform.
However, it cannot show earth becuase of some error on gdal lib etc. I'm trying to fix it now and contribute to the work. If Rocky plans to support it, the job of adapting those platforms will be easier, and solving some of the problems that I can't solve right now. 😄

**Reply by gwaldron** [reply, marked as answer]
OK great. GDAL is an optional support feature, so depending on the type of data you need to read you might be able to build without it. (For example, if you're just reading tiled maps from the internet.)
In any case we're happy to incorporate changes to support more platforms natively.

**Reply by GanAHE** [reply]
Thank you for your suggestion which gives me a better solution.  I will try to fix the error on the Android platform and sync the result on this discussion column.🥳
